### DISCUSSION TITLE
Larastan does not accept concrete model in place of Model

### DISCUSSION THREAD

**Discussion by wadakatu** [larastan/larastan#1201] [Q&A] [answered]
- PHP 7.4.28
- Laravel 6.20.44
- Larastan 1.0.3
- phpstan Level 3

Hello.
I have an error as below when I use `Illuminate\Database\Eloquent\Builder::when()` method.
However, I have no idea how to avoid this error.
Is there a way to resolve this error?

```
Anonymous function should return 
  Illuminate\Database\Eloquent\Builder<Illuminate\Database\Eloquent\Model> but returns
  Illuminate\Database\Eloquent\Builder<App\Http\Models\{MyModelName}>. 
```

My code is something like this.

```php
class MyModelName extends Model {

    public function getSomething() {
           return self::query()
                  ->select(['id', 'name'])
                  ->when(true, function (Builder $query) {
                           return $query->addSelect(DB::raw('month(start_at)')); 👈 
                  })
                  ->get();
    } 

}
```

Thank you.

**Comment by szepeviktor** [comment]
Hello! 👋 
Does you model extend `Illuminate\Database\Eloquent\Model`?

**Comment by szepeviktor** [comment]
```php
class MyModelName extends Model
```

Yes, it does.

**Comment by szepeviktor** [comment, marked as answer]
A 💎 diamond-level Laravel expert - an acquaintance of mine - told me to consider using a scope and cast it to the desired format.

**Reply by Gummibeer** [reply]
This doesn't sound like an answer to me. Replacing a custom query builder with scopes to make PHPStan happy sounds pretty bad to me. 🤔🤯

**Comment by wadakatu** [comment]
@szepeviktor 

Thank you for your reply.

I tried to use a scope and set Builder class as return type of the scope.
As a result, it worked and there is no error on console.

I appreciated you and your acquaintance answer.
You saved my day!
